lightning_client.go

```diff
@@ -1483,12 +1483,21 @@ func (s *lightningClient) payInvoice(ctx context.Context, invoice string,
 				)
 
 				r := payResp.PaymentRoute
-				preimage, err := lntypes.MakePreimage(
-					payResp.PaymentPreimage,
+
+				var (
+					preimage lntypes.Preimage
+					err      error
 				)
-				if err != nil {
-					return &PaymentResult{Err: err}
+
+				if payResp.PaymentPreimage != nil {
```

Comment:
should we maybe first check if this is actually an AMP invoice by checking `payReq.Features.IsSet(lnwire.AMPRequired)`? and then we only allow the pre-image to be nil if this is actually an AMP payment? 

perhaps that is overkill... lemme know what you think :)

Reply:
Discussed offline, want to avoid having as much logic in `lndclient` as possible.